provider/datagen/src/databake.rs

```diff
@@ -67,7 +75,18 @@ impl BakedDataExporter {
         {
             let mut file = crlify::BufWriterWithLineEndingFix::new(File::create(&path)?);
             writeln!(file, "// @generated")?;
-            writeln!(file, "{}", data)?;
+            if self.use_separate_crates {
+                writeln!(file, "{}", data)?;
+            } else {
+                writeln!(
+                    file,
+                    "{}",
+                    data.to_string()
+                        .replace("icu_", "icu :: ")
+                        .replace("icu :: provider", "icu_provider")
```

Comment:
Observation: this captures `icu_provider` and also `icu_provider_adapters`, which contains the fallback keys